quaddtype/tests/test_quaddtype.py

```diff
@@ -1,9 +1,22 @@
+import numpy as np
 
+from quaddtype import QuadDType, QuadScalar
 
-def test_instantiate():
-    """Test that importing the dtype works."""
-    import quaddtype  # noqa: F401
 
+def test_dtype_creation():
+    assert str(QuadDType()) == "This is a quad (128-bit float) dtype."
 
-if __name__ == '__main__':
-    test_instantiate()
+
+def test_scalar_creation():
+    assert str(QuadScalar(3.1)) == "3.1"
+
+
+def test_create_with_explicit_dtype():
+    assert repr(
+        np.array([3.0, 3.1, 3.2], dtype=QuadDType())
+    ) == "array([3.0, 3.1, 3.2], dtype=This is a quad (128-bit float) dtype.)"
```

Comment:
you could add a case that creates an array and calls `tobytes()` on it to verify that the data are actually 128 bit floats, since that's not visible from just looking at the repr

Reply:
:100: Excellent idea - see the new test I added.